### PR TITLE
refactor activate_convproc_method2

### PR DESCRIPTION
**Pull request by tangshuaiqi**
remove “method2” as there is no "method1" in the code

parameter use_cwaer_for_activate_maxsat is set as False, remove code related to it

separate out two subroutines: aer_vol_num_hygro, update_conu_from_act_frac

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 1]
```diff
@@ -2230,163 +2221,195 @@ subroutine ma_activate_convproc_method2(     &
...
+   real(r8) :: wdiab                  ! diabatic vertical velocity [m/s]
+   real(r8) :: wminf, wmaxf           ! limits for integration over updraft spectrum [m/s]
 
 !-----------------------------------------------------------------------
 
-
-
 ! check f_ent > 0
-   if (f_ent <= 0.0) return
+   if (f_ent <= 0.0_r8) return
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2269]: It is a good thing to add `_r8` but sometimes it does change answers and make them NBFB. So just be careful about it.

**Review by singhbalwinder — APPROVED** [review]
Looks great! I added a general comment but you don't need to change anything.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 [hunk 2]
```diff
@@ -2230,163 +2221,195 @@ subroutine ma_activate_convproc_method2(     &
...
-
-   real(r8), intent(in)    :: fracice ! Fraction of ice within the cloud
-                                     ! used as in-cloud wet removal rate
-   integer,  intent(in)    :: lun    ! logical unit for diagnostic output
-   integer,  intent(in)    :: lchnk  ! chunk index
-   integer,  intent(in)    :: i      ! column index
-   integer,  intent(in)    :: k      ! level index
+   integer, intent(in)     :: pcnst_extd        ! indices
+   ! conu = tracer mixing ratios in updraft at top of this (current) level. The conu are changed by activation
+   real(r8), intent(inout) :: conu(pcnst_extd)    ! [#/kg or kg/kg]
```
Comment (jeff-cohere) [inline components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2236]: Does this mean that `conu` refers to both number and mass mixing ratios, and that this subroutine can treat either type of quantity equivalently?

**Review by jeff-cohere — APPROVED** [review]
Looks good to me. I asked just one question to make sure I understand what's going on with `conu`.

File: components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90 (hunk 2 above)
Comment (tangshuaiqi) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2236]: yes... these variables have the first element (`ll == 0` in the old code) of number concentration for all aerosols and other elements (`do ll = 1, nspec_amode(n)` in the olde code) on mass concentration for each aerosol species.

Comment (singhbalwinder) [reply components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90:2236]: I am also making mental notes of how these appear in the code. We carry numbers and mmrs in different containers in the C++ code. We may need to restructure input to these routines or change the logic to comply with those data structures.

**Comment by tangshuaiqi** [comment]
All tests PASS. merge into maint

**Pull request merged**